### ISSUE TITLE
Error: Getting metadata from plugin failed with error: invalid_grant

### ISSUE THREAD

**Issue by ozgurgulsuna** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#58]
***
***You are Offline!***
***Please check your Internet Connection...***
***Error: Getting metadata from plugin failed with error: invalid_grant***

This is the error I get when I try to interact with it.
***
**Here is the console output**
```
[ozgur@ozgur ~]$ g-assist
Gtk-Message: 01:47:27.764: Failed to load module "colorreload-gtk-module"
WARNING: Kernel has no file descriptor comparison support: Operation not permitted
(node:2768) Electron: Loading non-context-aware native module in renderer: '/run/user/1000/snap.g-assist/.org.chromium.Chromium.p0zq7X'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:2830) Electron: Loading non-context-aware native module in renderer: '/run/user/1000/snap.g-assist/.org.chromium.Chromium.S8rREt'. This is deprecated, see https://github.com/electron/electron/issues/18397.
```
**Error on DevTools**
```
common.js: line 62 ;
    Auth error:Error: invalid_grant
main.js: Line 492 ;
    Error: 14 UNAVAILABLE: Getting metadata from plugin failed with error: invalid_grant
    at Object.exports.createStatusError (/snap/g-assist/1/res…pc/src/common.js:91)
    at ClientDuplexStream._emitStatusIfDone (/snap/g-assist/1/res…c/src/client.js:233)
    at ClientDuplexStream._receiveStatus (/snap/g-assist/1/res…c/src/client.js:211)
    at Object.onReceiveStatus (/snap/g-assist/1/res…nterceptors.js:1311)
    at InterceptingListener._callNext (/snap/g-assist/1/res…interceptors.js:568)
    at InterceptingListener.onReceiveStatus (/snap/g-assist/1/res…interceptors.js:618)
    at /snap/g-assist/1/res…nterceptors.js:1127
```
**Screenshots**
![Screenshot_20210104_015326](https://user-images.githubusercontent.com/38794995/103490853-bc07f680-4e2f-11eb-8236-c17aeee6f1c1.png)
and
![Screenshot_20210104_015503](https://user-images.githubusercontent.com/38794995/103490880-e48ff080-4e2f-11eb-9bfb-adb5965c9474.png)

**Desktop**
 - OS --> Host: ozgur Kernel: 5.8.11-1-MANJARO x86_64 bits: 64 Desktop: KDE Plasma 5.19.5 Distro: Manjaro Linux

**Comment by Melvin-Abraham** [comment]
@ozgurgulsuna You can fix this by deleting the tokens file that was fetched by the application (as per the _"Saved Tokens Path"_) and relaunch the assistant. You will again be presented with **"Get Token!"** screen. Authenticate your Gmail account again and let it fetch the tokens. This should possibly fix the issue...

**Comment by paperfairy** [comment]
@Melvin-Abraham Following these steps doesn't give me a new token, but it gives me a code. Where do we dump this code?

https://i.imgur.com/RA3H1y8.gif

**Comment by Melvin-Abraham** [comment]
@paperfairy This was a known issue (see #39). The patch will be released in the next release. For the time being this should help:

> Fixed in [dd67747](https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/commit/dd67747c068d5c70fd07cf823620cb24c7e6ab59). Since, this fix will not be readily available until next release, you can either clone this project and build it yourself or if you do not want to build it yourself, you can _temporarily_ disable **Settings** > **Conversation** > **Enable microphone on application startup**, relaunch assistant and let the assistant save the tokens after which you can enable the setting again.

_Ref: https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/issues/39#issuecomment-734111970_

**Comment by rodrigoswz** [comment]
> @ozgurgulsuna You can fix this by deleting the tokens file that was fetched by the application (as per the _"Saved Tokens Path"_) and relaunch the assistant. You will again be presented with **"Get Token!"** screen. Authenticate your Gmail account again and let it fetch the tokens. This should possibly fix the issue...

I was with the exact same problem, doing that fixed it. Thank you!

**Comment by ozgurgulsuna** [comment]
> @ozgurgulsuna You can fix this by deleting the tokens file that was fetched by the application (as per the _"Saved Tokens Path"_) and relaunch the assistant. You will again be presented with **"Get Token!"** screen. Authenticate your Gmail account again and let it fetch the tokens. This should possibly fix the issue...

doing that also fixed it for me, thanks a lot !

**Comment by kbdharun** [comment]
> @ozgurgulsuna You can fix this by deleting the tokens file that was fetched by the application (as per the _"Saved Tokens Path"_) and relaunch the assistant. You will again be presented with **"Get Token!"** screen. Authenticate your Gmail account again and let it fetch the tokens. This should possibly fix the issue...

Thanks, Sorry I didn't know this was already there, and I am Extremely Sorry for Duplicate Comment.

**Comment by Mantej-Singh** [comment]
> @ozgurgulsuna You can fix this by deleting the tokens file that was fetched by the application (as per the _"Saved Tokens Path"_) and relaunch the assistant. You will again be presented with **"Get Token!"** screen. Authenticate your Gmail account again and let it fetch the tokens. This should possibly fix the issue...

Thank you so much for porting this app on windows and also for a fix for "You are Offline!" issue

**Comment by CyanoTex** [comment]
> 
> 
> @paperfairy This was a known issue (see #39). The patch will be released in the next release. For the time being this should help:
> 
> > Fixed in [dd67747](https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/commit/dd67747c068d5c70fd07cf823620cb24c7e6ab59). Since, this fix will not be readily available until next release, you can either clone this project and build it yourself or if you do not want to build it yourself, you can _temporarily_ disable **Settings** > **Conversation** > **Enable microphone on application startup**, relaunch assistant and let the assistant save the tokens after which you can enable the setting again.
> 
> _Ref: [#39 (comment)](https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/issues/39#issuecomment-734111970)_
> 
> After following either of these steps, you will be able to continue with the **"Get Token"** screen without any issue. For additional help with saving the tokens, refer to [Guide: Setting authentication in the app](https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/wiki/Setup-Authentication-for-Google-Assistant-Unofficial-Desktop-Client#setting-authentication-in-the-app)

_**PLEASE**_ do not have your window behavior settings to Close on Blur. It can become extremely irritating when the issue pops up.

**Comment by anuragdandge** [comment]
how long do we have to wait after putting that code in "Get Token " Window ?
